packages/graph/hash_graph/lib/graph/src/api/rest/entity.rs

```diff
@@ -137,21 +132,39 @@ async fn get_entity<P: StorePool + Send>(
     tag = "Entity",
     responses(
         (status = 200, content_type = "application/json", description = "List of all entities", body = [PersistedEntity]),
+
         (status = 500, description = "Store error occurred"),
     )
 )]
 async fn get_latest_entities<P: StorePool + Send>(
     pool: Extension<Arc<P>>,
-    mut body: Option<Json<Expression>>,
 ) -> Result<Json<Vec<PersistedEntity>>, StatusCode> {
-    read_from_store(
-        pool.as_ref(),
-        &body
-            .take()
-            .map_or_else(Expression::for_latest_version, |json| json.0),
+    get_entities_by_query(pool, Json(Expression::for_latest_version())).await
+}
+
+#[utoipa::path(
+    get,
+    path = "/entities/{entityId}",
+    tag = "Entity",
+    responses(
+        (status = 200, content_type = "application/json", description = "The requested entity", body = PersistedEntity),
+        (status = 422, content_type = "text/plain", description = "Provided entity id is invalid"),
```

Comment:
How does this 422 happen now?

Reply:
I fixed the error code. This is an example output when passing an invalid entity id:
```
Invalid URL: UUID parsing failed: invalid character: expected an optional prefix of `urn:uuid:` followed by [0-9a-zA-Z], found `s` at 3
```
7bd4a83edc01ac71439853913470da9fe7fa7b10

Reply:
Does that correctly return a 400? I'm surprised because I don't see where we set the error code but perhaps that's just because I'm missing intellisense.

Reply:
Yes, it's returning 400 because `axum` will try to parse `EntityId`